### PR TITLE
Update key-figures-finland

### PR DESCRIPTION
**Pull request by tuopouk**
Updated the filename and ids

### CODE REVIEW

**Comment by AnnMarieW** [comment]
This looks good now and it's ready for Adam's review.  
Would you like to format it with Black?  If not we can do that after it's merged.

Thanks!

**Comment by tuopouk** [comment]
I just formatted it with Black.

**Review by Coding-with-Adam — COMMENTED** [review]
hi @tuopouk 
The code looks good to me. 

In efforts to limit the amount of data inside the assets folder on the Dash Example Index, I uploaded the json and csv files to [Plotly data sets repo](https://github.com/plotly/datasets/tree/master/Dash-Examle-Index). 

Then, I tried updating the `pd.read_csv()` on line 11 with the new [raw csv link](https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv), which should work. However, I could not load the [raw regions.json](https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/regions.json) file using `wtih open ... as` on line 37. Do any of you, @tuopouk @AnnMarieW, know how to read an http linked json file into json.loads?

I tried using some ideas from this [stackoverflow discussion](https://stackoverflow.com/questions/22676/how-to-download-a-file-over-http) but was not successful. 

My goal is to have all data files in the Plotly data sets repo to ensure that the [assets folder of this app](https://github.com/AnnMarieW/dash-app-gallery/tree/main/assets) is exclusively used for app images.

**Comment by tuopouk** [comment]
@Coding-with-Adam 
Maybe this could work...
`json.loads(json.dumps(requests.get(url).text)) `

File: examples/key-figures-finland.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    for stat in whole_country_df.index
+]
+
+# Drop since not needed.
+df.drop("WHOLE COUNTRY", axis=0, inplace=True)
+
+# The json file used for the mapbox viz.
+# https://geo.stat.fi/geoserver/wfs?service=WFS&version=2.0.0&request=GetFeature&typeName=maakunta1000k_2021&outputFormat=json
+with open("assets/regions.json", encoding="utf-8") as f:
+    regions_json = json.loads(f.read())
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:38]:
```suggestion
r = requests.get(r"https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/regions.json")
r.encoding='utf-8'
regions_json = r.json()
```

File: examples/key-figures-finland.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import pandas as pd
+from dash import Dash, dcc, html, Input, Output
+import dash_bootstrap_components as dbc
+import plotly.express as px
+import json
+
+# Data provided by Statistics Finland.
+# Original data source: https://pxdata.stat.fi/PxWeb/pxweb/en/Kuntien_avainluvut/Kuntien_avainluvut__2021/kuntien_avainluvut_2021_viimeisin.px/
+df = (
+    pd.read_csv("assets/key_figures_regions.csv", encoding="latin-1")
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:11]:
```suggestion
    pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv", encoding="latin-1")

```

File: examples/key-figures-finland.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+from dash import Dash, dcc, html, Input, Output
+import dash_bootstrap_components as dbc
+import plotly.express as px
+import json
+
+# Data provided by Statistics Finland.
+# Original data source: https://pxdata.stat.fi/PxWeb/pxweb/en/Kuntien_avainluvut/Kuntien_avainluvut__2021/kuntien_avainluvut_2021_viimeisin.px/
+df = (
+    pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv", encoding="latin-1")
+
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:12]:
```suggestion
```

File: examples/key-figures-finland.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+import pandas as pd
+from dash import Dash, dcc, html, Input, Output
+import dash_bootstrap_components as dbc
+import plotly.express as px
+import json
+
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:7]:
```suggestion
import requests

```

**Comment by tuopouk** [comment]
@Coding-with-Adam 
Could you please change the reading csv part to have encoding to `utf-8`? If it's not the same encoding, some regions will not be rendered on the map.
`df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv', encoding="utf-8").rename(columns={"Region 2021": "Region"}).set_index("Region")`

File: examples/key-figures-finland.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+from dash import Dash, dcc, html, Input, Output
+import dash_bootstrap_components as dbc
+import plotly.express as px
+import json
+import requests
+
+# Data provided by Statistics Finland.
+# Original data source: https://pxdata.stat.fi/PxWeb/pxweb/en/Kuntien_avainluvut/Kuntien_avainluvut__2021/kuntien_avainluvut_2021_viimeisin.px/
+df = (
+    pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv", encoding="latin-1")
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:12]:
```suggestion
    pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv", encoding="utf-8")

```

File: examples/key-figures-finland.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+import dash_bootstrap_components as dbc
+import plotly.express as px
+import json
+import requests
+
+# Data provided by Statistics Finland.
+# Original data source: https://pxdata.stat.fi/PxWeb/pxweb/en/Kuntien_avainluvut/Kuntien_avainluvut__2021/kuntien_avainluvut_2021_viimeisin.px/
+df = (
+    pd.read_csv("https://raw.githubusercontent.com/plotly/datasets/master/Dash-Examle-Index/key_figures_regions.csv", encoding="utf-8")
+
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:13]:
```suggestion
```

File: examples/key-figures-finland.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    # Use space as thousand separator.
+    stat_value = "{:,}".format(stat_value).replace(",", " ")
+
+    return html.Div(
+        [
+            html.Div([dff.stat_name, ", ", dff.year]),
+            html.Div("in Finland"),
+            html.Span([stat_value, " ", dff.unit]),
+        ],
+    )
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:125]:
```suggestion
    return html.Div(
        [
            html.Div([dff.stat_name, ", ", dff.year]),
            html.Div("in Finland"),
            html.Span([f"{stat_value:,}", " ", dff.unit]),
        ],
    )

```

File: examples/key-figures-finland.py [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    stat_value = {True: int(dff.value), False: dff.value}[".0" in str(dff.value)]
+
+    return html.Div(
+        [
+            html.Div([dff.stat_name, ", ", dff.year]),
+            html.Div("in Finland"),
+            html.Span([f"{stat_value:,}", " ", dff.unit]),
+        ],
+    )
+
```
Comment (Coding-with-Adam) [inline examples/key-figures-finland.py:123]:
```suggestion
```

**Comment by Coding-with-Adam** [comment]
> , encoding="utf-8").rename(columns={"Region 2021": "Region"}).set_index("Region")

Done. Sorry I missed that. 

@tuopouk 
I also tweaked the line of code that originally replaced the comma with a space. This was a suggestion by Ann that I agreed with.

Let me know if you like the final code/result. And I will merge the app.

**Comment by tuopouk** [comment]
@Coding-with-Adam 
Looks great and the regions render correctly as well as their names. It's actually better to have comma as a separator. It would be different if the app was non-English. I guess you can merge from my part. Thanks!

**Comment by Coding-with-Adam** [comment]
Thanks for adding that file back, @tuopouk

**Review by Coding-with-Adam — APPROVED** [review]

**Pull request merged**